File: app/src/main/java/com/commit451/gitlab/model/api/Pipelines.java
package com.commit451.gitlab.model.api;

import com.squareup.moshi.Json;

import org.parceler.Parcel;

/**
 * Represents a pipeline
 */
@Parcel
public class Pipelines {

    @Json(name = "sha")
    String sha;
    @Json(name = "id")
    long id;
    @Json(name = "ref")
    String ref;
    @Json(name = "status")
    String status;

    public long getId() {
        return id;
    }

    public String getSha() {
        return sha;
    }

    public String getRef() {
        return ref;
    }

    public String getStatus() {
        return status;
    }
}

File: app/src/main/java/com/commit451/gitlab/model/api/Pipelines.kt
package com.commit451.gitlab.model.api

import com.squareup.moshi.Json

import org.parceler.Parcel

/**
 * Represents a pipeline
 */
@Parcel
class Pipelines {

    @field:Json(name = "sha")
    var sha: String? = null
    @field:Json(name = "id")
    var id: Long = 0
    @field:Json(name = "ref")
    var ref: String? = null
    @field:Json(name = "status")
    var status: String? = null
}
